osu.Game/Online/Chat/DrawableLinkCompiler.cs

```diff
@@ -52,6 +62,32 @@ private void load(OsuColour colours)
 
         protected override IEnumerable<Drawable> EffectTargets => Parts;
 
+        public MenuItem[] ContextMenuItems
+        {
+            get
+            {
+                List<MenuItem> items = new List<MenuItem>();
+                bool hasTooltip = TooltipText.ToString() != "";
+
+                string text = hasTooltip ? TooltipText.ToString() : GetUrlFromPart(Parts);
+
+                bool isChannelorLobbyLink = text.Contains("osu://chan") || text.Contains("osump://");
+
+                if (!isChannelorLobbyLink)
```

Comment:
Doesn't need to exist if what I'm saying above is true.

Reply:
Hmm, you make a good point. I'll try splitting it off the compiler.